Resfeber/View Controllers/ExploreViewController.swift

```diff
@@ -134,6 +134,46 @@ extension ExploreViewController: WaterfallLayoutDelegate {
     }
 }
 
+// MARK: - Collection View Delegate
+
+extension ExploreViewController: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, contextMenuConfigurationForItemAt indexPath: IndexPath, point: CGPoint) -> UIContextMenuConfiguration? {
+        guard let cell = collectionView.cellForItem(at: indexPath) as? DestinationCell,
+              let destination = cell.destination else { return nil }
+        
+        return UIContextMenuConfiguration(identifier: nil, previewProvider: nil) { suggestedActions in
+            
+            // Setup Favorite menu item
+            
+            let addToFavoritesText = destination.isFavorite ? "Remove Favorite" : "Favorite"
+            
+            let favoriteImage: UIImage?
+            if destination.isFavorite {
+                favoriteImage = UIImage(systemName: "heart.fill")?.withTintColor(.systemRed, renderingMode: .alwaysOriginal)
+            } else {
+                favoriteImage = UIImage(systemName: "heart")
+            }
+            
+            let addToFavorites = UIAction(title: addToFavoritesText, image: favoriteImage) { action in
```

Comment:
You probably want the closure to be a weak reference `[weak self] action in`

Reply:
This closure does not need to capture self (see my response to your comment on lines 164-167)

Reply:
Added `[weak self]` in PR #18